R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+#' `'nm74gf'`, `'nm74gf_nmfe'`, and `'nm75'`.
+#'  - `nmtran_presort` is an supplementary utility that preprocesses the control
+#'     stream to ensure it is in the correct format for `NM-TRAN`. It is particularly
+#'     relevant for handling specific data manipulations and ensuring compatibility
+#'     with the `NM-TRAN` executable.
+#'
+#' @param .mod A `bbi_nonmem_model` object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` flags are passed to `NM-TRAN`.
+#'   See [print_bbi_args()] for more details.
+#' @inheritParams submit_model
+#' @param .nonmem_version Character scalar for default version of NONMEM to use.
+#'   If left `NULL`, will look for the default version specified in the provided
+#'   `bbi` configuration file.
+#' @param clean Logical (`T`/`F`). If `FALSE`, don't delete the temporary folder
+#'   containing the `NM-TRAN` run, which will be stored in the current working
+#'   directory.
+#' @param run_dir Directory to run `NM-TRAN` in. Only relevant if `clean = FALSE`.
+#'
+#' @examples
+#' \dontrun{
+#'
+#' mod <- read_model(file.path(MODEL_DIR, 1))
+#' run_nmtran(mod, .nonmem_version = "nm75")
+#'
+#' # Save the run directory for manual inspection
+#' run_nmtran(mod, clean = FALSE, run_dir = getwd())
+#'
+#' }
+#'
+#' @export
+run_nmtran <- function(
+    .mod,
+    .bbi_args = list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2),
+    .nonmem_version = NULL,
+    .config_path = NULL,
+    run_dir = tempdir(),
+    clean = TRUE
+){
+  check_model_object(.mod, "bbi_nonmem_model")
+
+  # Capture NONMEM/NM-TRAN options and format `NM-TRAN` args
+  nmtran_specs <- nmtran_setup(.mod, .nonmem_version, .bbi_args, .config_path)
+
+  mod_path <- get_model_path(.mod)
+  data_path <- get_data_path_from_ctl(.mod)
+
+  # Make temporary directory in current directory
+  mod_name <- fs::path_ext_remove(basename(mod_path))
+  temp_folder <- withr::local_tempdir(
+    pattern = paste0("nmtran-mod_", mod_name, "-"),
+    tmpdir = run_dir, clean = clean
+  )
+
+  # Copy model
+  file.copy(mod_path, temp_folder)
+  nmtran_mod <- new_model(file.path(temp_folder, basename(mod_path)))
+
+  # Copy dataset & overwrite $DATA record of new model
+  # NM-TRAN will error if data cannot be found
+  if(fs::file_exists(data_path)){
+    file.copy(data_path, temp_folder)
+    # overwrite $DATA record of new model
+    modify_data_path_ctl(nmtran_mod, data_path = basename(data_path))
+  }
+
+  # Run NM-TRAN
+  nmtran_results <- c(
+    list(
+      nmtran_exe = nmtran_specs$nmtran_exe,
+      nmtran_presort_exe = nmtran_specs$nmtran_presort_exe,
+      nonmem_version = nmtran_specs$nonmem_version,
+      absolute_model_path = mod_path
+    ),
+    execute_nmtran(
+      nmtran_specs$nmtran_exe,
+      mod_path = basename(mod_path),
```

Comment:
[ This is an implementation detail for an internal function that can be left as is, but fwiw I find it confusing. ]

`mod_path` is absolute here, and you take call `basename` because `execute_nmtran` is designed to take that. Yet, in all the spots `execute_nmtran` uses `mod_path`, it reconstructs that full path by joining the passed directory. Why design it to take the `basename` then?